RestCommClient/Classes/RCConnection.m

```diff
@@ -299,6 +301,32 @@ - (void)sipManagerDidReceiveOutgoingDeclined:(SipManager*)sipManager;
         [self.callingPlayer stop];
         self.callingPlayer.currentTime = 0.0;
     }
+    
+    // We want to tell to delegate that connection is declined immediately
+    // so, we need to play busy tone i new thread because we dont know what client will do
+    // for example, if set instance on connection to nil, the sound will not be played
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+        NSError *error;
+        AVAudioPlayer * busyPlayer;
+        // busy
+        NSString *filename = @"busy_tone_sample.mp3";
+        // we are assuming the extension will always be the last 3 letters of the filename
+        NSString *file = [[NSBundle mainBundle] pathForResource:[filename substringToIndex:[filename length] - 3 - 1]
+                                                         ofType:[filename substringFromIndex:[filename length] - 3]];
+        
+        if (file) {
+            busyPlayer = [[AVAudioPlayer alloc] initWithContentsOfURL:[NSURL fileURLWithPath:file] error:&error];
+            if (!busyPlayer) {
+                NSLog(@"Error: %@", [error description]);
+                return;
+            }
+            
+        }
+        
+        [busyPlayer play];
+        sleep(2);
```

Comment:
Maybe we can replace the 2 seconds literal with AVAudioPlayer.duration, so that we 're sure the full duration of the mp3 is played even if we use a different mp3, wdyt?

Reply:
Yes, I think we should do it like it.